src/components/metadataFormControls/ModelSingleSelect/ModelSingleSelect.tsx

```diff
@@ -84,6 +87,16 @@ export const ModelSingleSelect = <
         return transform ? transform(allDataMap) : allDataMap
     }, [allDataMap, transform])
 
+    useEffect(() => {
+        // fetch until we have resolvedAvailable length equal to the desired page size (or until we reach the end)
+        if (
+            resolvedAvailable.length < (queryObject.params.pageSize ?? 10) &&
+            queryResult.hasNextPage
```

Comment:
I think it would be on the safe side and add a guard for `!isFetching` to prevent out-of-order request, or potential duplicate requests. Or maybe just `queryResult.fetchNextPage({ cancelRefetch: false})` would have the same behavior?

Reply:
used `fetchNextPage({ cancelRefetch: false})` as I think that should be the same behaviour and doesn't add a new dependency